src/libfetchers/github.cc

```diff
@@ -332,9 +333,10 @@ struct GitLabInputScheme : GitArchiveInputScheme
     Hash getRevFromRef(nix::ref<Store> store, const Input & input) const override
     {
         auto host = maybeGetStrAttr(input.attrs, "host").value_or("gitlab.com");
+        auto repo = std::regex_replace(getStrAttr(input.attrs, "repo"), std::regex("/"), "%2F");
```

Comment:
There's a urlencode function in libutil: https://github.com/NixOS/nix/blob/master/src/libutil/url.cc#L94

Probably should just use that.

Reply:
That would be far better than regex, I'll switch over.

Reply:
Ive tested and it works as intended, switched over now.